app/models/project_media.rb

```diff
@@ -430,6 +430,31 @@ def apply_rules_and_actions_on_update
     self.team.apply_rules_and_actions(self, rule_ids)
   end
 
+  def self.handle_fact_check_for_existing_claim(existing_pm,new_pm)
+    if existing_pm.fact_check.blank?
+      existing_pm.append_fact_check_from(new_pm)
+    elsif existing_pm.fact_check.present?
+      if existing_pm.fact_check.language != new_pm.set_fact_check['language']
+        new_pm.replace_with_blank_media
+      end
+    end
+  end
+
+  def append_fact_check_from(new_pm)
+    self.set_claim_description = new_pm.set_claim_description
+    self.set_fact_check = new_pm.set_fact_check
+    self.create_claim_description_and_fact_check
+    self
+  end
+
+  def replace_with_blank_media
+    m = Blank.create!
+    self.set_original_claim = nil
+    self.media_id = m.id
+    self.save!
+    self
+  end
```

Comment:
Suggestion for this whole block... I think it's more clear to:

- Explicitly say, after line 435, that we're returning the existing item: `return existing_pm`
- Explicitly say, after line 438, that we're returning the new item: `return new_pm`

And then in the methods `append_fact_check_from` and `replace_with_blank_media`, you don't need to return `self` in their last lines (447 and 455).